docs/dev/developer-guides/transactions/fee-model.md

```diff
@@ -39,3 +36,67 @@ So the tips to make the most out of the zkSync fee system are the following:
+We provide a convenient way for anyone to estimate the cost of a transaction regardless of the type of account. 
+### Changes in the `validateTransaction`
+
+Right now, the `validateTransaction` fails if the signature is not correct and returns `success=true` otherwise, while the `returndata` is discarded completely. We will migrate to a different approach, where the `validateTransaction` method to be considered successful must return some magic string. Any other `returndata` will be considered invalid and will be rejected by the system.  
+
+Now, `validateTransaction` method of the AA (or the paymaster `validateAndPayForPaymasterTransaction` method), should always try to do the same amount of computation (and especially storage accesses) as it would have done if the transaction was indeed validated correctly. By default, the operator will provide a transaction structure that contains as much information as is available to the operator at the time of the fee estimation. As a substitute for the signature, an invalid 65-byte ECDSA signature will be used. The `DefaultAccount` (the one that is used by EOAs) will do as many operations as possible (including signature verification) and only return `bytes4(0)` instead of magic. If, for instance, the wallet is a custom account, with multiple signers, then it may want to simulate the validation of the signatures of all the provided signers.
+
+So roughly the code of the validation step of each account will look the following way:
+
+```solidity
+ function validateTransaction(
+        bytes32, // _txHash
+        bytes32 _suggestedSignedHash,
+        Transaction calldata _transaction
+    ) external payable override ignoreNonBootloader returns (bytes4 magic) {
+        // ...
+        // normal validation logic
+        // ...
+         
+        // The execution before the following `if` statement should look as close as to the execution under incorrect 
+        // signature as possible
+        if(!validSignature) {
+           // Setting magic to 0 to make sure that the transaction will not be accepted on mainnet
+           magic = bytes4(0);  
+        } else {
+           magic = // setting to actual magic value which should be returned by the correct signature validation
+        }
+    }
+```
+
+### Notes on the transaction’s length
+
+zkSync Era sends state diffs onchain, and the cost for the transaction will still mildly depend on its length (because long transactions need to be stored in memory of the operator), also long transactions incur additional costs during interactions with an account. However, the signature (as well as its length) is not available at the time of fee estimation and so there is no correct way to precisely estimate the cost of the transaction. For now, we will compensate for it by multiplying the recommended cost of the transaction by **5%**. In the future, we will introduce the following:
```

Comment:
Let's not hardcode "5%" here. It is better to give us some room to change it without updating the documentation. Better to say "by a few percents".

Also, it is better not to say "we will introduce the following" (I am sorry if it was present in the doc this way). We should rather say "may", because we do not commit to it, though it is likely what may happen in the future